jupyterhub_files/spawner.py

```diff
@@ -221,34 +231,34 @@ def check_for_hanged_ec2(self, instance):
     def poll(self):
         """ Polls for whether process is running. If running, return None. If not running,
             return exit code """
-        self.log.debug("function poll for user %s" % self.user.name)
+        self.log_user("poll()")
         try:
             instance = yield self.get_instance()
-            self.log.debug(instance.state)
+            self.log_user("poll: instance state is %s" % instance.state)
             if instance.state['Name'] == 'running':
-                self.log.debug("poll: server is running for user %s" % self.user.name)
+                self.log_user("poll: instance is running, checking...")
                 # We cannot have this be a long timeout because Jupyterhub uses poll to determine whether a user can log in.
                 # If this has a long timeout, logging in without notebook running takes a long time.
                 # attempts = 30 if self.notebook_should_be_running else 1
                 # check if the machine is hanged
                 ec2_run_status = yield self.check_for_hanged_ec2(instance)
                 if ec2_run_status == "SSH_CONNECTION_FAILED":
-                    #self.log.debug(ec2_run_status)
+                    self.log_user("poll: instance is hanging: %s" % ec2_run_status)
                     yield self.kill_instance(instance)
                     return "Instance Hang"
                 else:
                     notebook_running = yield self.is_notebook_running(instance.private_ip_address, attempts=1)
                     if notebook_running:
-                        self.log.debug("poll: notebook is running for user %s" % self.user.name)
+                        self.log_user("poll: notebook is running")
                         return None #its up!
                     else:
-                        self.log.debug("Poll, notebook is not running for user %s" % self.user.name)
+                        self.log_user("poll: notebook is NOT running") 
                         return "server up, no instance running for user %s" % self.user.name
```

Comment:
Hi @arthurian, I just noticed a possible typo here. It should probably say "Server up, no **notebook** running for user ..." instead of "Server up, no **instance** running for user...". This was there from before, not introduced by you. Just pointing it out.

Reply:
Yeah I think you're right about that.